ext/odbc/php_odbc_includes.h

```diff
@@ -240,8 +246,8 @@ ZEND_BEGIN_MODULE_GLOBALS(odbc)
     zend_long default_cursortype;
     char laststate[6];
     char lasterrormsg[SQL_MAX_MESSAGE_LENGTH];
-	HashTable *resource_list;
-	HashTable *resource_plist;
+	HashTable results;
+	HashTable connections;
```

Comment:
I originally put it there AFAIR but I changed for `GSHUTDOWN` after this review comment: https://github.com/php/php-src/pull/12040#discussion_r1313993791 I even looked up https://www.phpinternalsbook.com/php7/extensions_design/php_lifecycle.html#globals-termination-gshutdown and thought that `GSHUTDOWN` is called during `RSHUTDOWN` assuming that these globals should be freed there, however, I missed this sentence at the very end of the section: `Remember that globals are not cleared after every request; aka GSHUTDOWN() is not called as part of RSHUTDOWN().` So you are surely right, I'll use `RSHUTDOWN` then.

Reply:
> As a side question, why do we hold such a reference at all? I know this is pre-existing, but I think it makes sense to clean up non-persistent connections as soon as the connection object becomes unreachable instead of waiting until the request ends, no?

The `connections` hash table is needed because there's a `odbc_close_all()` function which closes all connections right away. The `results` hash table is needed because results are automatically closed as soon as a connection is closed... And I agree with you, I also think that `odbc_close_all()` is probably the main source of complexity in this extension... Also, by having this list in place, it's possible to implement connection reuse if we want to (it is already available for persistent resources).